fuel/src/main/kotlin/com/github/kittinunf/fuel/core/DataPart.kt

```diff
@@ -1,9 +1,223 @@
+ * @param contentType [String] the content-type of the inline content; defaults to text/plain; charset=utf-8
+ * @param contentDisposition [String] defaults to form-data with name and filename set, unless filename is null
+ */
+data class InlineDataPart(
+    val content: String,
+    val name: String,
+    val fileName: String? = null,
+    override val contentType: String = "$GENERIC_CONTENT; charset=utf-8",
+    override val contentDisposition: String = "form-data; name=\"$name\"${if (fileName != null) "; filename=\"$fileName\"" else "" }"
+) : DataPart() {
+    override fun inputStream() = content.byteInputStream()
+    override val contentLength get() = content.length.toLong()
+}
+
+/**
+ *  DataPart that represents a file
+ *
+ *  @example Add a file as a DataPart
+ *
+ *    val file = File("foo.json")
+ *    request.add { FileDataPart(file, name = "field-name", fileName = "remote-file-name.json") }
+ *
+ *    // ...boundary
+ *    // Content-Disposition: "form-data; name=metadata; filename=remote-file-name.json"
+ *    // Content-Type: "application/json"
+ *    //
+ *    // <file-contents foo.json>
+ *
+ *  @example Adding multiple files under the same field name, formerly using multipart/mixed:
+ *
+ *    val file1 = File("foobar.json")
+ *    val file2 = File("baz.json")
+ *    request.add(FileDataPart(file1, name = "field-name"), FileDataPart(file2, name = "field-name"))
+ *
+ *    // ...boundary
+ *    // Content-Disposition: "form-data; name=field-name; filename=foobar.json"
+ *    // Content-Type: "application/json"
+ *    //
+ *    // <file-contents foobar.json>
+ *    //
+ *    //
+ *    // ...boundary
+ *    // Content-Disposition: "form-data; name=field-name; filename=baz.json"
+ *    // Content-Type: "application/json"
+ *    //
+ *    // <file-contents baz.json>
+ *
+ *  @example Adding multiple files as an array to a single field name:
+ *
+ *    val file1 = File("foobar.json")
+ *    val file2 = File("baz.json")
+ *    request.add(FileDataPart(file1, name = "field-name[]"), FileDataPart(file2, name = "field-name[]"))
+ *
+ *    // ...boundary
+ *    // Content-Disposition: "form-data; name=field-name[]; filename=foobar.json"
+ *    // Content-Type: "application/json"
+ *    //
+ *    // <file-contents foobar.json>
+ *    //
+ *    //
+ *    // ...boundary
+ *    // Content-Disposition: "form-data; name=field-name[]; filename=baz.json"
+ *    // Content-Type: "application/json"
+ *    //
+ *    // <file-contents baz.json>
+ *
+ *  @param file [File] the source file
+ *  @param name [String] the field name, defaults to the file name without extension
+ *  @param fileName [String] the remote file name, or null
+ *  @param contentType [String] the content type of the file contents; defaults to a guess using [guessContentType]
+ *  @param contentDisposition [String] defaults to form-data with name and filename set, unless filename is null
+ */
+data class FileDataPart(
     val file: File,
```

Comment:
[Question] As this is a `FileDataPart`, do you think we can add another API in such a way that we can skip passing `File` object altogether for the sake of ease of use? For example, In the test the usage could be

```kotlin
request.add(FileDataPart(System.getProperty("user.dir").resolve("src/test/assets"), "lorem_ipsum_long.tmp"))
```

I think the usage of passing a "String" or a path that indicates the location of the file should be the most ubiquitous?

Reply:
Ah, I think that is a reasonable approach! 👍

Reply:
This now works.